verifier/check_known_issues.py

```diff
@@ -28,125 +33,142 @@
+
+    # sm_opcodes describe the change to turn expected string into the actual string
+    # See https://docs.python.org/3/library/difflib.html#difflib.SequenceMatcher.get_opcodes
+    sm = SequenceMatcher(None, expected, actual)
+    sm_opcodes = sm.get_opcodes()
+
+    digit_replace = False
+    non_digit_replacement = False
+
+    # sm_opcodes describe the changes to turn expected string into the actual string
+    # See https://docs.python.org/3/library/difflib.html#difflib.SequenceMatcher.get_opcodes
+    # The tuple is [tag, i1, i2, j1, k2]
+    # Tag indicates the type of change.
+    # i1:i2 is the range of the substring in expected
+    # j1:j2 is the range of the substring in actual
+
+    for diff in sm_opcodes:
+        tag = diff[0]  # 'replace', 'delete', 'insert', or 'equal'
+        old_val = expected[diff[1]:diff[2]]
+        new_val = actual[diff[1]:diff[2]]
+        if tag == 'replace':
+            # expected[i1:i2] was replaced by actual[j1:j2]
+            if old_val.isdigit() and new_val.isdigit():
+                # TODO!! : check the value of the numeral
+                # If the same value, then its a numbering system difference
+                if unicodedata.numeric(old_val) == unicodedata.numeric(new_val):
+                    digit_replace = True
                 else:
-                    # a digit was replaced with a non-digit
+                    # Both were digits but different numeric values
                     non_digit_replacement = True
+            else:
+                # a digit was replaced with a non-digit
+                non_digit_replacement = True
+
+    # Only true if the only changes were replacing digits
+    if digit_replace and not non_digit_replace:
+        return knownIssueType.known_issue_replaced_numerals
+    else:
+        return None
+
+
+def check_datetime_test(test_data, known_issues_list):
+    # Examine a single test for date/time isses
+    # Returns known issues identified for this test in this category
+    remove_this_one = False
+    try:
+        result = test_data['result']
+        expected = test_data['expected']
+        is_ki = diff_nbsp_vs_sp(result, expected)
+        if is_ki:
+            # Mark the test with this issue
+            test_data['known_issue'] = knownIssueType.known_issue_nbsp_sp.value
+            remove_this_one = True
+
+        is_ki = numerals_replaced(result, expected)
+        if is_ki:
+            test_data['known_issue_id'] = knownIssueType.known_issue_replaced_numerals.value
+            remove_this_one = True
+
+    except BaseException as err:
+        # Can't get the info
+        pass
+        pass
+
+    if remove_this_one:
+        known_issues_list.append(test_data)
+
+    return remove_this_one
 
-        # Only true if the only changes were replacing digits
-        return digit_replace and not non_digit_replace
 
-def check_datetime(test_results):
+def check_issues(test_type, test_results_to_check):
```

Comment:
reminder: maybe organize code from high-level at the bottom of the file to low-level code at the top of the file